packages/kbn-tinymath/test/library.test.js

```diff
@@ -31,96 +43,134 @@ describe('Parser', () => {
-      expect(check('" foo bar"')).toThrow('but "\\"" found');
-      expect(check('"foo bar "')).toThrow('but "\\"" found');
-      expect(check('"0foo"')).toThrow('but "\\"" found');
-    });
-
-    it('invalid characters in single quotes', () => {
-      const check = (str) => () => parse(str);
-      /* eslint-disable prettier/prettier */
-      expect(check("' foo bar'")).toThrow('but "\'" found');
-      expect(check("'foo bar '")).toThrow('but "\'" found');
-      expect(check("'0foo'")).toThrow('but "\'" found');
-      expect(check("' foo bar'")).toThrow('but "\'" found');
-      expect(check("'foo bar '")).toThrow('but "\'" found');
-      expect(check("'0foo'")).toThrow('but "\'" found');
-      /* eslint-enable prettier/prettier */
+      expect(parse('"_foo bar"')).toEqual(variableEqual('_foo bar'));
+      expect(parse('"@foo bar"')).toEqual(variableEqual('@foo bar'));
+      expect(parse('".foo bar"')).toEqual(variableEqual('.foo bar'));
+      expect(parse('"-foo bar"')).toEqual(variableEqual('-foo bar'));
+      expect(parse('"_foo0 bar1"')).toEqual(variableEqual('_foo0 bar1'));
+      expect(parse('"@foo0 bar1"')).toEqual(variableEqual('@foo0 bar1'));
+      expect(parse('".foo0 bar1"')).toEqual(variableEqual('.foo0 bar1'));
+      expect(parse('"-foo0 bar1"')).toEqual(variableEqual('-foo0 bar1'));
+      expect(parse('" foo bar"')).toEqual(variableEqual(' foo bar'));
+      expect(parse('"foo bar "')).toEqual(variableEqual('foo bar '));
+      expect(parse('"0foo"')).toEqual(variableEqual('0foo'));
+      expect(parse('" foo bar"')).toEqual(variableEqual(' foo bar'));
+      expect(parse('"foo bar "')).toEqual(variableEqual('foo bar '));
+      expect(parse('"0foo"')).toEqual(variableEqual('0foo'));
     });
   });
 
   describe('Functions', () => {
     it('no arguments', () => {
-      expect(parse('foo()')).toEqual({ name: 'foo', args: [] });
+      expect(parse('foo()')).toEqual(functionEqual('foo', []));
     });
 
     it('arguments', () => {
-      expect(parse('foo(5,10)')).toEqual({ name: 'foo', args: [5, 10] });
+      expect(parse('foo(5,10)')).toEqual(functionEqual('foo', [5, 10]));
     });
 
     it('arguments with strings', () => {
-      expect(parse('foo("string with spaces")')).toEqual({
-        name: 'foo',
-        args: ['string with spaces'],
-      });
+      expect(parse('foo("string with spaces")')).toEqual(
+        functionEqual('foo', [variableEqual('string with spaces')])
+      );
 
-      /* eslint-disable prettier/prettier */
-      expect(parse("foo('string with spaces')")).toEqual({
-        name: 'foo',
-        args: ['string with spaces'],
-      });
-      /* eslint-enable prettier/prettier */
+      expect(parse("foo('string with spaces')")).toEqual(
+        functionEqual('foo', [variableEqual('string with spaces')])
+      );
+    });
+
+    it('named only', () => {
+      expect(parse('foo(q=10)')).toEqual(functionEqual('foo', [namedArgumentEqual('q', '10')]));
```

Comment:
It seems like named parameters are always strings which is a little limiting (any reason we don't do numbers here?). I guess we could do the parsing in Lens though if necessary.

If that's the case, can we add a test for dots in the named parameter?

Reply:
@flash1293 I had a draft where I allowed any sub-expression as a named argument, but the main reason I chose to make these literal strings is that it creates an ambiguous grammar. Every other string in Tinymath is treated as a variable and throws an error if it isn't passed into the evaluation context. So for example "1d" is a constant value that has special meaning in Elasticsearch, but Tinymath would throw errors at parsing time because it's not a variable from the context.

So making these literal strings simplifies the grammar.

Reply:
Thanks for explaining! In that case, could we add some tests whether common number formats work? e.g. `abc(q=0.23)`